sdk/daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala

```diff
@@ -152,6 +151,7 @@ class Engine(val config: EngineConfig = Engine.StableConfig, allowLF2: Boolean =
     for {
       pkgResolution <- preprocessor.buildPackageResolution(packageMap, packagePreference)
       processedCmds <- preprocessor.preprocessApiCommands(pkgResolution, cmds.commands)
+      _ <- preprocessor.prefetchKeys(processedCmds)
       processedDiscs <- preprocessor.preprocessDisclosedContracts(disclosures)
```

Comment:
dummy question, should we not avoid prefetching key from disclosure ?

Reply:
Good point.